### ISSUE TITLE
Feature ask: sibling version of bucket script aggregation

### ISSUE THREAD

**Issue by monfera** [elastic/elasticsearch#52522]
**Describe the feature**:

[Bucket script aggregation](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-pipeline-bucket-script-aggregation.html) is a parent pipeline aggregation and it provides flexibility for when the premade pipeline aggregations won't cover some need. 

For symmetry, and to meet the data requirements of a broad range of of visualizations, it'd be useful to make it alternatively work as a sibling aggregation.

Here are a couple of visualizations (I'm not suggesting that these can't be done with other, more targeted sibling aggregations or with distinct, N-ary parent aggregations, so they just illustrate the structure of the data need):

Heatmap with marginal histograms: the histograms are just aggregations (along two different fields) of the heatmap part
![image](https://user-images.githubusercontent.com/1548516/74853558-62ddc400-533e-11ea-9e84-aaffd3b8ac1b.png)

Scatterplot with boxplot: the boxplot height/position etc. can be computed as statistics over the scatterplot points (which themselves may be aggregates)
![image](https://user-images.githubusercontent.com/1548516/74853812-ba7c2f80-533e-11ea-870f-00c3df39db86.png)

Basically, any kind of multilayer charts; eg. here the faint points are the most granular aggregates, and everything else including the bars are aggregations over them, this is a random prototype for a multilayer visualization (I'm not sure if the exponentially weighted moving average line and the Bollinger band around it is feasible, but the others likely are)
![image](https://user-images.githubusercontent.com/1548516/74854123-2b234c00-533f-11ea-814d-6fab45b444b7.png)

Entered on encouragement by @polyfractal
